jenkins-design-language/src/js/components/material-ui/SvgIcon/SvgIcon.js

```diff
@@ -0,0 +1,102 @@
+import React, {Component, PropTypes } from 'react';
+
+class SvgIcon extends Component {
+
+  static propTypes = {
+    /**
+     * Elements passed into the SVG Icon.
+     */
+    children: PropTypes.node,
+    /**
+     * This is the fill color of the svg icon.
+     * If not specified, this component will default
+     * to muiTheme.palette.textColor.
+     */
+    color: PropTypes.string,
+    /**
+     * This is the icon color when the mouse hovers over the icon.
+     */
+    hoverColor: PropTypes.string,
+    /** @ignore */
+    onMouseEnter: PropTypes.func,
+    /** @ignore */
+    onMouseLeave: PropTypes.func,
+    /**
+     * Override the inline-styles of the root element.
+     */
+    size: PropTypes.number,
+    style: PropTypes.object,
+    /**
+     * Allows you to redefine what the coordinates
+     * without units mean inside an svg element. For example,
+     * if the SVG element is 500 (width) by 200 (height), and you
+     * pass viewBox="0 0 50 20", this means that the coordinates inside
+     * the svg will go from the top left corner (0,0) to bottom right (50,20)
+     * and each unit will be worth 10px.
+     */
+    viewBox: PropTypes.string,
+  };
+
+  static defaultProps = {
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    viewBox: '0 0 24 24',
+  }; 
+
+  state = {
+    hovered: false,
```

Comment:
I kept it like this because if we send the fill as a prop, shouldn't we be able to send the hover color as well? I think it gives us an extra layer of flexibility so we can make the icon look specifically how we want without any extra css